src/ast/analysis/Type.cpp

```diff
@@ -522,6 +529,249 @@ void TypeAnalysis::run(const TranslationUnit& translationUnit) {
+}
+
+void TypeAnnotationPrinter::print_(type_identity<StringConstant>, const StringConstant& cnst) {
+    os << cnst << "∈{string}";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<NumericConstant>, const NumericConstant& constant) {
+    os << constant;
+    auto numericConstantTypes = typeAnalysis.getNumericConstantTypes();
+    auto numType = numericConstantTypes.find(&constant)->second;
+    if (numType == NumericConstant::Type::Int) {
+        os << "∈{Int}";
+    } else if (numType == NumericConstant::Type::Uint) {
+        os << "∈{Uint}";
+    } else if (numType == NumericConstant::Type::Float) {
+        os << "∈{Float}";
+    } else {
+        os << "∈{???}";
+    }
+}
+
+void TypeAnnotationPrinter::print_(type_identity<BinaryConstraint>, const BinaryConstraint& rel) {
+    auto lhs = rel.getLHS();
+    auto lTySet = argumentTypes.find(lhs)->second;
+    assert(lTySet.size() == 1);
+    auto lTy = lTySet.begin();
+    branchOnArgument(lhs, *lTy);
+    if (isA<RecordInit>(lhs) || isA<UnnamedVariable>(lhs)) {
+        os << "∈{" << lTy->getName() << "}";
+    }
+
+    os << " " << rel.getBaseOperator() << " ";
+
+    auto rhs = rel.getRHS();
+    auto rTySet = argumentTypes.find(rhs)->second;
+    assert(rTySet.size() == 1);
+    auto rTy = rTySet.begin();
+    branchOnArgument(rhs, *rTy);
+    if (isA<RecordInit>(rhs) || isA<UnnamedVariable>(rhs)) {
+        os << "∈{" << rTy->getName() << "}";
+    }
+}
+
+void TypeAnnotationPrinter::print_(type_identity<IntrinsicFunctor>, const IntrinsicFunctor& fun) {
+    auto arguments = fun.getArguments();
+    if (arguments.size() == 2) {  // binary
+        auto tySet = argumentTypes.find(arguments[0])->second;
+        assert(tySet.size() == 1);
+        auto ty = tySet.begin();
+        os << "(";
+        branchOnArgument(arguments[0], *ty);
+        if (isA<RecordInit>(arguments[0]) || isA<UnnamedVariable>(arguments[0])) {
+            os << "∈{" << ty->getName() << "}";
+        }
+
+        os << " " << fun.getBaseFunctionOp() << " ";
+
+        auto tySet2 = argumentTypes.find(arguments[1])->second;
+        assert(tySet2.size() == 1);
+        auto ty2 = tySet2.begin();
+        branchOnArgument(arguments[1], *ty2);
+        if (isA<RecordInit>(arguments[1]) || isA<UnnamedVariable>(arguments[1])) {
+            os << "∈{" << ty2->getName() << "}";
+        }
+
+        os << ")";
+    } else {
+        os << fun.getBaseFunctionOp() << "(";
+        for (std::size_t i = 0; i < arguments.size(); ++i) {
+            TypeAttribute argType = typeAnalysis.getFunctorParamTypeAttribute(fun, i);
+            auto& ty = typeEnv.getConstantType(argType);
+            branchOnArgument(arguments[i], ty);
+            if (i + 1 < arguments.size()) {
+                os << ",";
+            }
+        }
+        os << ")";
+    }
+}
+
+void TypeAnnotationPrinter::print_(
+        type_identity<UserDefinedFunctor>, [[maybe_unused]] const UserDefinedFunctor& fun) {
+    // TODO
```

Comment:
Can we add the printing for User-Defined Functor (perhaps we add the signature??)?

Reply:
The function has been added.